core/src/main/java/com/emc/ia/sdk/support/http/HttpClient.java

```diff
@@ -29,7 +29,7 @@
     return post(uri, headers, type, payload);
   }
 
-  <T> T post(String uri, Collection<Header> headers, Class<T> type, Part... parts) throws IOException;
+  <T> T post(String uri, Collection<Header> headers, Class<T> type, Collection<Part> parts) throws IOException;
```

Comment:
I don't like it too, but there's a reason of this changes. In `uploadTransformationFile()` it should add only one Part to post request. In this case it would be called method `post(String uri, Class<T> type, S payload)`, but it's wrong. So, Ray, how to avoid this issue?